internal/commands/run.go

```diff
@@ -37,36 +38,58 @@ func runCmd(dockerCli command.Cli) *cobra.Command {
 		Short:   "Run an application",
 		Long:    longDescription,
 		Example: example,
-		Args:    cli.ExactArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return runRun(dockerCli, args[0], opts)
+			if opts.cnabBundle != "" {
+				return runCnab(dockerCli, opts)
+			}
+			if len(args) != 1 {
```

Comment:
I think the test should be first checking if cnab flag AND the arg are both set.

Reply:
I'm not sure I understand